### PR TITLE
477 allow all domains

### PR DESCRIPTION
**Pull request by UlrichKu**
| Q             | A
| ------------- | ---
| Branch        | 6.2.x
| Bug fix?      | yes
| New feature?  | yes - sort of: is mostly fixes a buggy behavior
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed issues  | chameleon-system/chameleon-system#477
| License       | MIT

### CODE REVIEW

File: src/CoreBundle/EventListener/AllowEmbeddingForDifferentDomainListener.php [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+class AllowEmbeddingForDifferentDomainListener
+{
+    /**
+     * @param GetResponseEvent $event
+     */
+    public function onKernelRequest(GetResponseEvent $event)
+    {
+        $request = $event->getRequest();
+
+        if (true === $this->canAllowAdditionalDomains($request)) {
```
Comment (rnuernberg) [inline src/CoreBundle/EventListener/AllowEmbeddingForDifferentDomainListener.php]: why "can" and not just `allowAdditionalDomains`?

Comment (rnuernberg) [reply src/CoreBundle/EventListener/AllowEmbeddingForDifferentDomainListener.php]: i would also prefer early exit here to keep the method simple.

File: src/CoreBundle/EventListener/AllowEmbeddingForDifferentDomainListener.php [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+            $refererHost = $this->getRefererHost();
+            if (null !== $refererHost && $request->getHost() !== $refererHost) {
```
Comment (rnuernberg) [inline src/CoreBundle/EventListener/AllowEmbeddingForDifferentDomainListener.php]: again, we could keep it a bit simpler by using early exit
```php
if (false === $this->allowAdditionalDomains($request) {

    return ;
}
$refererHost = $this->getRefererHost();
if (null === $refererHost || $request->getHost() === $refererHost) {

    return ;
}

header("X-Frame-Options: ALLOW-FROM $refererHost");
```

File: src/CoreBundle/EventListener/AllowEmbeddingForDifferentDomainListener.php [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+        // TODO the priority in service definition as low as it is is necessary for CMSUserDefined() to work..
+
+        return 'true' === $request->get('__previewmode') && true === \TGlobalBase::CMSUserDefined();
+    }
+
+    private function getRefererHost(): ?string
+    {
+        // TODO can the referer be determined more high-level than $_SERVER?
+
+        if (false === \array_key_exists('HTTP_REFERER', $_SERVER)) {
```
Comment (rnuernberg) [inline src/CoreBundle/EventListener/AllowEmbeddingForDifferentDomainListener.php]: why not get this from the request? 
```php
$request->server->get('HTTP_REFERER')
```

File: src/CoreBundle/EventListener/RequestTypeListener.php [hunk 4]
```diff
@@ -62,6 +62,7 @@ public function onKernelRequest(GetResponseEvent $event)
         $request = $event->getRequest();
         $requestType = $this->getRequestType();
         $requestType->initialize();
+
```
Comment (rnuernberg) [inline src/CoreBundle/EventListener/RequestTypeListener.php]: any reason for the added line?

File: src/CoreBundle/EventListener/AllowEmbeddingForDifferentDomainListener.php (hunk 1 above)
Comment (UlrichKu) [reply src/CoreBundle/EventListener/AllowEmbeddingForDifferentDomainListener.php]: ok

File: src/CoreBundle/EventListener/AllowEmbeddingForDifferentDomainListener.php (hunk 2 above)
Comment (UlrichKu) [reply src/CoreBundle/EventListener/AllowEmbeddingForDifferentDomainListener.php]: ok

File: src/CoreBundle/EventListener/RequestTypeListener.php (hunk 4 above)
Comment (UlrichKu) [reply src/CoreBundle/EventListener/RequestTypeListener.php]: removed

File: src/CoreBundle/EventListener/AllowEmbeddingForDifferentDomainListener.php (hunk 3 above)
Comment (UlrichKu) [reply src/CoreBundle/EventListener/AllowEmbeddingForDifferentDomainListener.php]: ok

**Comment by UlrichKu** [comment]
Changed logic from trying to use the session to using any configured domain name.

**Pull request merged**